### DISCUSSION TITLE
Trunctate table using prisma

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [prisma/prisma#22013] [Q&A] [answered]
### Question

Hello ,
how can i truncate a table using prisma
Is it possible to do it in prisma studio like send a sql query or like in cli without touching the code ?
Thank you !

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @Elmatbaisaad 👋 

Thank you for raising this question.

Truncating a table using Prisma can be done by writing raw SQL queries and executing them with Prisma's raw database access methods. See this section of the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/crud#deleting-all-data-with-raw-sql--truncate). There is also an open [issue](https://github.com/prisma/prisma/issues/5063) on using truncating a table with a documented workaround using code.

Prisma Studio does not currently support sending raw SQL queries directly through the UI. If you need to perform a truncate operation without touching the code, you would typically use a database management tool that connects directly to your database and allows you to execute raw SQL queries.

If you prefer to use the CLI, Prisma provides a way to execute SQL files or SQL commands from the command line. For example, you can execute a SQL file using the following command:
```
prisma db execute --file ./script.sql --schema schema.prisma
```
Or you can execute a SQL command directly from the command line like this:
```
echo 'TRUNCATE TABLE dev;' | prisma db execute --stdin --url="$DATABASE_URL"
```

See [documentation](https://www.prisma.io/docs/reference/api-reference/command-reference#db-execute) on using `db execute` command.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
